### PR TITLE
CFE-748: Update CoreDNS EgressFirewall integration enhancement proposal

### PR DESCRIPTION
**Pull request by arkadeepsen**
Update EP with the location of the DNSNameResolver controller code. Update the EP with the latest API details from https://github.com/openshift/api/pull/1524.

### CODE REVIEW

File: enhancements/dns/coredns-egressfirewall-integration.md [hunk 1]
```diff
@@ -128,20 +128,23 @@ type DNSNameResolver struct {
 	Status DNSNameResolverStatus `json:"status,omitempty"`
 }
 
+// DNSName is used for validation of a DNS name.
+// +kubebuilder:validation:Pattern=`^(\*\.)?([a-z0-9]([-a-z0-9]{0,61}[a-z0-9])?\.){2,}$`
+// +kubebuilder:validation:MaxLength=254
```
Comment (npinaeva) [inline enhancements/dns/coredns-egressfirewall-integration.md:133]: validation changes are coming from the api PR feedback right?

Comment (arkadeepsen) [reply enhancements/dns/coredns-egressfirewall-integration.md:133]: > validation changes are coming from the api PR feedback right?

Yes.

File: enhancements/dns/coredns-egressfirewall-integration.md [hunk 2]
```diff
@@ -549,11 +558,11 @@ also validates that a label in a DNS name does not start or end with a `-`.
...
 	// ..
 
-	// dnsName is the domain name to allow/deny traffic to. If this is set, cidrSelector must be unset.
+	// dnsName is the domain name to allow/deny traffic to. If this is set, cidrSelector and nodeSelector must be unset.
 	// For a wildcard DNS name, the '*' will match only one label. Additionally, only a single '*' can be
 	// used at the beginning of the wildcard DNS name. For example, '*.example.com' will match 'sub1.example.com'
-	// but won't match 'sub2.sub1.example.com'
-	// +kubebuilder:validation:Pattern=^(\*\.)?([A-Za-z0-9]([-A-Za-z0-9]*[A-Za-z0-9])?\.)*[A-Za-z0-9]([-A-Za-z0-9]*[A-Za-z0-9])?\.?$
+	// but won't match 'sub2.sub1.example.com'.
+	// +kubebuilder:validation:Pattern=`^(\*\.)?([A-Za-z0-9-]+\.)*[A-Za-z0-9-]+\.?$`
```
Comment (npinaeva) [inline enhancements/dns/coredns-egressfirewall-integration.md:565]: where is this pattern coming from? IIUC it is different from the DNSName validation

Comment (arkadeepsen) [reply enhancements/dns/coredns-egressfirewall-integration.md:565]: The pattern is from the upstream ovnk PR: https://github.com/ovn-org/ovn-kubernetes/pull/4045/files#diff-afeb50cd661258df2d05c385a5779e7e33dad7e4caf85be278b190fb1a22fd7bR84

**Comment by huiran0826** [comment]
@arkadeepsen One question regarding ` only allow rule scenarios have been used in this enhancement proposal.`,  does it mean it will not support Deny rule for dnsName any more in egressFirewall ?   cc @npinaeva

**Comment by arkadeepsen** [comment]
> @arkadeepsen One question regarding ` only allow rule scenarios have been used in this enhancement proposal.`, does it mean it will not support Deny rule for dnsName any more in egressFirewall ? cc @npinaeva

Yes @huiran0826. However, the EgressFirewall CRD will not be changed for this and though it'll be possible to create Deny DNS rules using this feature, the behavior may not be as expected.

**Comment by npinaeva** [comment]
Right, for egress firewall we probably will stick to the current behaviour, but using DNS names for deny rules never made a lot of sense. We may need to update our docs to outline that

**Pull request merged**